src/stcal/resample/resample.py

```diff
@@ -4,6 +4,7 @@
 import sys
 
 import numpy as np
+from collections import defaultdict
```

Comment:
```suggestion
from collections import defaultdict

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from numpy.typing import DTypeLike
```

isort/ruff will likely want to change this to move the check after the imports but I can't make a suggestion on those lines.